src/algorithms/reco/JetReconstruction.cc

```diff
@@ -57,41 +58,48 @@ namespace eicrecon {
     // Print out some infos
     m_log->trace("  Clustering with : {}", jet_def.description());
 
+    // loop over jets
     for (unsigned i = 0; i < jets.size(); i++) {
 
       m_log->trace("  jet {}: pt = {}, y = {}, phi = {}", i, jets[i].pt(), jets[i].rap(), jets[i].phi());
-      edm4eic::MutableReconstructedParticle jet_edm;
+
       // Type = 0 for jets, Type = 1 for constituents
       // Use PDG values to match jets and constituents
-      jet_edm.setType(0);
-      jet_edm.setPDG(i);
-      jet_edm.setMomentum(edm4hep::Vector3f(jets[i].px(), jets[i].py(), jets[i].pz()));
-      jet_edm.setEnergy(jets[i].e());
-      jet_edm.setMass(jets[i].m());
-
+      edm4eic::MutableReconstructedParticle jet_output;
+      jet_output.setType(0);
+      jet_output.setPDG(i);
+      jet_output.setMomentum(edm4hep::Vector3f(jets[i].px(), jets[i].py(), jets[i].pz()));
+      jet_output.setEnergy(jets[i].e());
+      jet_output.setMass(jets[i].m());
+
+      // loop over constituents
       std::vector<PseudoJet> csts = jets[i].constituents();
       for (unsigned j = 0; j < csts.size(); j++) {
+
         const double cst_pt = csts[j].pt();
         m_log->trace("    constituent {}'s pt: {}", j, cst_pt);
 
-        edm4eic::MutableReconstructedParticle cst_edm;
         // Type = 0 for jets, Type = 1 for constituents
         // Use PDG values to match jets and constituents
-        cst_edm.setType(1);
-        cst_edm.setPDG(i);
-        cst_edm.setMomentum(edm4hep::Vector3f(csts[j].px(), csts[j].py(), csts[j].pz()));
-        cst_edm.setEnergy(csts[j].e());
-        cst_edm.setMass(csts[j].m());
-        //jet_edm.addToParticles(cst_edm);  // FIXME: global issue with podio reference
+        edm4eic::MutableReconstructedParticle cst_output;
+        cst_output.setType(1);
+        cst_output.setPDG(i);
+        cst_output.setMomentum(edm4hep::Vector3f(csts[j].px(), csts[j].py(), csts[j].pz()));
+        cst_output.setEnergy(csts[j].e());
+        cst_output.setMass(csts[j].m());
+        //jet_output.addToParticles(cst_output);  // FIXME: global issue with podio reference
```

Comment:
This one would need to be routed properly, something along the lines of
```c++
add_to_particle(jet_output, input_collection[csts[j].user_index()])
```
where
```c++
template <typename T>
add_to_particle(edm4eic::MutableReconstructedParticle &p, T &other);

template <>
add_to_particle(edm4eic::MutableReconstructedParticle &p, edm4eic::ReconstructedParticle &other) {
  p.addToParticles(other);
}

template <>
add_to_particle(edm4eic::MutableReconstructedParticle &p, edm4eic::MCParticle &other) {
  p.addToMCParticles(other);
}
```